### ISSUE TITLE
Rowdetails Filtered list scrollbar doesn't appear on expand

### ISSUE THREAD

**Issue by SatanEnglish** [6pac/SlickGrid#221]
When list is filtered. (Say to 10 items)

The grid scroll bar doesn't show when an item is expanded.

Therefore you can no longer scroll to items under the expanded view

![2018-03-20_18-19-41](https://user-images.githubusercontent.com/2100563/37637216-4c6241c0-2c6b-11e8-9102-95b19b1644e2.gif)

**Comment by SatanEnglish** [comment]
@6pac @ghiscoding 
Hay guys any idea where I would look at to try and fix this issue. (Show stopper)
If I cannot figure it out i'm like to have to go use a different grid.

Must be some restraint that not there when the filter isn't applied.

**Comment by ghiscoding** [comment]
Really hard to say without a project to troubleshoot with. Though it might be related to the row detail being created on a row that doesn't exist anymore since it got filtered. Did the girl say she had issues with filtering on the original article?

I can't help much without a project, I haven't started anything with this row detail yet.

**Comment by SatanEnglish** [comment]
@ghiscoding 

Found the issue we may need to add some notes to our plugin somewhere.
The issue is how you have to filter. 

Filter was checking data off `dataView().getItem(0)["FieldX"].`

However in order to encapsulate the search with the detail view an extra check is required.

Example:
    // Set the item to search as the row item
    var itemToSearch =dataView().getItem(0)
    
    // If we are using the plugin and the _parent is not null (Defined in function getPaddingItem) 
    // we are actually looking at the an expanded row there filter using the parent
    if (args.hasDetailViewPlugin && itemToSearch['_parent'] != undefined) {
        itemToSearch = item._parent;
    }
    // Now search the item
    if (filterVal != itemToSearch["FieldX"]) {
          found = false;
          break;
     }  

So basically the issue was although I looked like I had the item as it showed.
In reality the expanding view rows were not included in the filtered list and therefore the grid space wan't growing to accommodate it.

**Comment by 6pac** [comment]
That's definitely worth adding to the example page filter, and noting in the page content.

**Comment by SatanEnglish** [comment]
@ghiscoding @6pac 
Another thing on this filtering that I've just found out.

This is multipling you filter time and if using something like moment.js which is slow for string formatted dates the ordering becomes very slow.

**Comment by SatanEnglish** [comment]
If you interested I ended up parsing it the first time round into epoch then putting it into another property and filtering of that this way it only needs to calculate the date once for parent as its there by the time you hide the child.
